salt/returners/__init__.py

```diff
@@ -164,7 +164,7 @@ def _options_browser(cfg, ret_config, defaults, virtualname, options):
         # default place for the option in the config
         value = _fetch_option(cfg, ret_config, virtualname, options[option])
 
-        if value:
+        if value != "":
```

Comment:
I suspect `value is not None` also should be in this condition, but I am not sure what the the previous condition was trying to achieve.

Reply:
The previous condition try to avoid parameters without value and then, not record them in the dict. But the previous code catchs parameters with "False" value which souldn't be, causing thoose parameters to be excluded of the dict.

Reply:
`value is not None` would miss `value = ""`. The original code `if value` would evaluate `True` for both scenarios(`None` and `""`). Not sure what this code is actually fixing.

Reply:
Hello twangboy,

Have you read the issue https://github.com/saltstack/salt/issues/66816 ? 
"if value:" doesn't work for any configuration directives that is equal to False, like this 

```
highstate.report_failures: True
highstate.report_changes: False
highstate.report_everything: False
```

Which gives this logs :

`[DEBUG   ] highstate setup {'report_format': 'html', 'report_delivery': 'smtp', 'smtp_sender': 'salt@myhost.com', 'smtp_recipients': 'ext.guillaume@noname.fr', 'smtp_failure_subject': 'failure minion {id} on host {host}', 'smtp_success_subject': 'succes minion {id} on host {host}', 'smtp_server': 'smtpunix.it.noname.fr'}`

Fixing code with "if value != """ gives this log :

`[DEBUG   ] highstate setup {'report_everything': False, 'report_changes': False, 'report_failures': True, 'report_format': 'html', 'report_delivery': 'smtp', 'smtp_sender': 'salt@myhost.com', 'smtp_recipients': 'ext.guillaume@noname.fr', 'smtp_failure_subject': 'failure minion {id} on host {host}', 'smtp_success_subject': 'succes minion {id} on host {host}', 'smtp_server': 'smtpunix.it.noname.fr'}`

This proceed reports to be send as desired. 

Regards.

Reply:
OK, so we're just checking if value is set to either ``True`` or ``False``. Do we need to handle ``None`` here?

This will also need a changelog and some tests. You can you ``caplog`` to test debug output.